pkg/controller/datavolume/controller-base.go

```diff
@@ -429,13 +430,14 @@ func (r *ReconcilerBase) syncDvPvcState(log logr.Logger, req reconcile.Request,
 		return syncState, err
 	}
 
-	if dv.DeletionTimestamp != nil {
-		log.Info("DataVolume marked for deletion, cleaning up")
-		if cleanup != nil {
-			if err := cleanup(&syncState); err != nil {
-				return syncState, err
-			}
+	if cleanup != nil {
+		if err := cleanup(&syncState); err != nil {
+			return syncState, err
 		}
+	}
+
+	if dv.DeletionTimestamp != nil {
+		log.Info("DataVolume marked for deletion")
```

Comment:
can you provide some context for these changes?  seems weird to always call cleanup.  To me cleanup should only be called when the operation completes or the deleted while in progress.  If ther is come other callback required for populators, maybe make it explicit?

Reply:
please read the first commit message